### ISSUE TITLE
Custom resource types can't be validated with --override-spec

### ISSUE THREAD

**Issue by medmunds** [awslabs/cfn-python-lint#445]
*cfn-lint version:* 0.8.3

*Description of issue:* When using custom resource types and supplying an override spec for the custom type, no property validations seem to be performed.

Example: specify a `Custom::Example` type with a required `ID` property and an optional Boolean `Enabled` property:

test.cf.yaml:
```yaml
Resources:
  GoodExample:
    Type: Custom::Example
    Properties:
      ServiceToken: example.arn
      ID: good
      Enabled: true

  BadExample:
    Type: Custom::Example
    Properties:
      ServiceToken: example.arn
      # Missing required prop `ID`
      Enabled: [invalid, 'must be a boolean']
      Wrong: 'this prop is not in the spec'
```

CustomExampleSpec.json:
```json
{
  "ResourceTypes": {
    "Custom::Example": {
      "Properties": {
        "ServiceToken": {"PrimitiveType": "String", "Required": true, "UpdateType": "Immutable"},
        "ID": {"PrimitiveType": "String", "Required": true, "UpdateType": "Immutable"},
        "Enabled": {"PrimitiveType": "Boolean", "Required": false, "UpdateType": "Mutable"}
      }
    }
  },
  "ResourceSpecificationVersion": "2.11.0"
}
```

Run: `cfn-lint --override-spec CustomExampleSpec.json test.cf.yaml`

Results: No lint errors or warnings

Expected:
```
E3003 Property ID missing at Resources/BadExample/Properties
test.cf.yaml:11:5

E3002 Property should be of type Boolean not List at Resources/BadExample/Properties/Enabled
test.cf.yaml:14:7

E3002 Invalid Property Resources/BadExample/Properties/Wrong
test.cf.yaml:15:7
```

This is occurring because #295 unconditionally forces any type starting with `Custom::` to `AWS::CloudFormation::CustomResource`, which deliberately allows all properties. For unknown custom types, that's incredibly useful. But if you have a known spec for the custom type, it would be good to use that instead.

I think this can be accomplished by changing `if resourcetype.startswith('Custom::')` to add `... and not resourcetype in self.resourcetypes` in both [Properties.py](https://github.com/awslabs/cfn-python-lint/blob/v0.8.3/src/cfnlint/rules/resources/properties/Properties.py#L270) and [Required.py](https://github.com/awslabs/cfn-python-lint/blob/v0.8.3/src/cfnlint/rules/resources/properties/Required.py#L132).

**Comment by cmmeyer** [comment]
Oooh. I don’t think it ever occurred to us that people could add specs for custom resources, but that makes perfect sense.

Any chance you’d want to take a stab at a PR? If you’re uncomfortable testing against your custom spec, we could certainly gin up a fake one.

**Comment by medmunds** [comment]
@cmmeyer here you go.
